scipy/stats/_stats_py.py

```diff
@@ -10165,6 +10165,136 @@ def first_order(t):
+    order = np.arange(1, 5) if order is None else np.asarray(order)
+    if not np.issubdtype(order.dtype, np.integer) or np.any(order <= 0):
+        raise ValueError(message)
+
+    axis = np.asarray(axis)[()]
+    message = "`axis` must be an integer."
+    if not np.issubdtype(axis.dtype, np.integer) or axis.ndim != 0:
+        raise ValueError(message)
+
+    sorted = np.asarray(sorted)[()]
+    message = "`sorted` must be True or False."
+    if not np.issubdtype(sorted.dtype, np.bool_) or sorted.ndim != 0:
+        raise ValueError(message)
+
+    standardize = np.asarray(standardize)[()]
+    message = "`standardize` must be True or False."
+    if not np.issubdtype(standardize.dtype, np.bool_) or standardize.ndim != 0:
+        raise ValueError(message)
+
+    sample = np.moveaxis(sample, axis, -1)
+    sample = np.sort(sample, axis=-1) if not sorted else sample
+
+    return sample, order, axis, sorted, standardize
+
+
+def _br(x, *, r=0):
+    n = x.shape[-1]
+    x = np.expand_dims(x, axis=-2)
+    x = np.broadcast_to(x, x.shape[:-2] + (len(r), n))
+    x = np.triu(x)
+    j = np.arange(n, dtype=x.dtype)
+    n = np.asarray(n, dtype=x.dtype)[()]
+    return (np.sum(special.binom(j, r[:, np.newaxis])*x, axis=-1)
+            / special.binom(n-1, r) / n)
+
+
+def _prk(r, k):
+    return (-1)**(r-k)*special.binom(r, k)*special.binom(r+k, k)
+
+
+@_axis_nan_policy_factory(  # noqa: E302
+    _moment_result_object, n_samples=1, result_to_tuple=lambda x: (x,),
+    n_outputs=lambda kwds: _moment_outputs(kwds, [1, 2, 3, 4])
+)
+def lmoment(sample, order=None, *, axis=0, sorted=False, standardize=True):
+    r"""Compute L-moments of a sample from a continuous distribution
+
+    The L-moments of a probability distribution are summary statistics with
+    uses similar to those of conventional moments, but they are defined in
+    terms of the expected values of order statistics.
+    Sample L-moments are defined analogously to population L-moments, and
+    they can serve as estimators of population L-moments. They tend to be less
+    sensitive to extreme observations than conventional moments.
+
+    Parameters
+    ----------
+    sample : array_like
+        The real-valued sample whose L-moments are desired.
+    order : array_like, optional
+        The (positive integer) orders of the desired L-moments.
+        Must be a scalar or non-empty 1D array. Default is [1, 2, 3, 4].
+    axis : int, default=0
+        The axis along which to compute L-moments.
+    sorted : bool, default=False
+        Whether `sample` is already sorted in increasing order along `axis`.
+        If False (default), `sample` will be sorted.
+    standardize : bool, default=True
+        Whether to return L-moment ratios for orders 3 and higher.
+        L-moment ratios are analogous to standardized conventional
+        moments: they are the non-standardized L-moments divided
+        by the L-moment of order 2.
```

Comment:
Wikipedia also claims that the first L-moment is often referred to as the "L-mean", which I've never once read in the dozens of papers that I've read about it 🤷🏻.

Reply:
Besides Wikipedia, "standardize" is used by Hosking.

![image](https://github.com/user-attachments/assets/06815a47-3567-4622-b8c7-95348191d6ce)

Conceptually, this means something very similar to "standardizing" in the normal sense (pun intended), so I think I'd prefer to leave it alone. This would also go nicely with a similar argument added to `scipy.stats.moment`.

I'll leave this comment unresolved so @rlucas7 can take a look, though.

Reply:
Haha ok you got me there. 

But he never used the term "standardized L-moments" as far as I'm aware. 
And the way that he used the term "standardize" there was done very carefully, as he immediately explains what exactly he means by it, i.e.

> [...] so that they are independent of the units of measurement of $X$.

But besides the "standard" deviation there's also the ["standard score"](https://en.wikipedia.org/wiki/Standard_score), "standardization", and the "standard normal distribution" (just to name a few) that are all defined in terms of the standard deviation.

However, now that I think about it, that's not always the case:
The "standard" uniform distribution, for example, has a standard deviation of $\sqrt{1 / 12}$ (and L-scale of $1 / 6$).
And the "standard" logistic distribution has a standard deviation of $\pi / \sqrt{3}$ (and L-scale of $1$ (yet another win for L-moments)):

```pycon
>>> import lmo
>>> import scipy.stats
>>> scipy.stats.norm.std(), scipy.stats.norm.l_scale()
(np.float64(1.0), 0.5641895835477564)
>>> scipy.stats.uniform.std(), scipy.stats.uniform.l_scale()
(np.float64(0.28867513459481287), 0.16666666666666666)
>>> scipy.stats.logistic.std(), scipy.stats.logistic.l_scale()
(np.float64(1.8137993642342178), 1.0)
```

Reply:
> And the way that he used the term "standardize" there was done very carefully, as he immediately explains what exactly he means by it,

In the documentation for `standardize`, the first sentence identifies them as the L-moment ratios. Just like he did, I'm using the term for familiarity, and immediately describing what I mean.

Reply:
> In the documentation for `standardize`

I'm not sure what you're talking about here 🤔